maker/src/routes.rs

```diff
@@ -1,52 +1,70 @@
+use crate::bitmex::Quote;
+use anyhow::Result;
 use http_api_problem::HttpApiProblem;
+use http_api_problem::StatusCode;
 use rocket::serde::json::Json;
 use rocket::serde::Deserialize;
 use rocket::serde::Serialize;
-
-#[derive(Serialize, Deserialize, Debug)]
-pub struct Price(u64);
-
-impl From<u64> for Price {
-    fn from(val: u64) -> Self {
-        Self(val)
-    }
-}
+use rocket::State;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
+use tokio::sync::watch;
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct Offer {
-    bid: Price,
-    ask: Price,
+    bid: u64,
+    ask: u64,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct Offers {
     long: Offer,
     short: Offer,
-    index_price: Price,
+    index_price: u64,
 }
 
 #[rocket::get("/offers")]
-pub async fn get_offers() -> Result<Json<Offers>, HttpApiProblem> {
-    // TODO: Use real values instead of hard-coding
+pub async fn get_offers(
+    rx_quote_receiver: &State<watch::Receiver<Option<Quote>>>,
+) -> Result<Json<Offers>, HttpApiProblem> {
+    let rx_quote_receiver = rx_quote_receiver.inner().clone();
+    let quote = *rx_quote_receiver.borrow();
 
-    let long = Offer {
-        bid: 19000.into(),
-        ask: 21000.into(),
-    };
+    // TODO: take spread from clap
+    let spread = dec!(0.015);
 
-    // Different values only to ensure we're fetching the correct ones
-    let short = Offer {
-        bid: 20500.into(),
-        ask: 19500.into(),
-    };
+    match quote {
+        Some(quote) => {
+            let long = Offer {
+                bid: (quote.bid * (Decimal::ONE - spread))
+                    .round()
+                    .to_u64()
+                    .expect("to fit into u64"),
+                ask: quote.ask.round().to_u64().expect("to fit into u64"),
+            };
 
-    let index_price = 19750.into();
+            let short = Offer {
+                bid: quote.bid.round().to_u64().expect("to fit into u64"),
+                ask: (quote.ask * (Decimal::ONE + spread))
+                    .round()
+                    .to_u64()
+                    .expect("to fit into u64"),
+            };
```

Comment:
Why do we have 2 offers each with it's own bid/ask pair?

Reply:
I think it has to be either one offer with bid and ask, that is then mapped to long and short, or two offers with one price where one offer representing short and one long. The later is what we had in ItchySats. In ItchySats we opted for two offers because we had other parameters that would depend on the position (long/short); but the model was always confusing.

As @bonomat has stated in https://github.com/itchysats/10101/pull/197#pullrequestreview-1181644762 it might be easier to have one offer with bid and ask price for now, and then the UI derives what to display from that offer, i.e. we have one offer with the maker's bid and ask price which represents what the maker is offering to the taker - i.e. the bid represents the price for going short (selling) and the ask for going long (buying). @bonomat please validate that this makes sense ;)

Reply:
One more note for clarification:

![image](https://user-images.githubusercontent.com/5557790/202041231-d688f87f-ba8c-432b-9d9f-d3d2d19b1910.png)

Bid and ask represent short and long and should not change when we switch between long and short in the UI. In ItchySats the price changed because it was part of the switch, i.e. we never displayed both prices in ItchySats but either the one for going long or short.

Reply:
fixed 4c34e9813bfceeb710430ccbf3fd7780978eaa41